operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakRealmJob.java

```diff
@@ -0,0 +1,206 @@
+        super(client, realmCR);
+        this.realmCR = realmCR;
+        this.secretName = secretName;
+        this.volumeName = KubernetesResourceUtil.sanitizeName(secretName + "-volume");
+
+        this.existingJob = fetchExistingJob();
+        this.existingDeployment = fetchExistingDeployment();
+    }
+
+    @Override
+    protected Optional<HasMetadata> getReconciledResource() {
+        if (existingJob == null) {
+            Log.info("Creating a new Job");
+            return Optional.of(createImportJob());
+        } else {
+            Log.info("Job already available");
+            return Optional.empty();
+        }
+    }
+
+    private Job fetchExistingJob() {
+        return client
+                .batch()
+                .v1()
+                .jobs()
+                .inNamespace(getNamespace())
+                .withName(getName())
+                .get();
+    }
+
+    private Deployment fetchExistingDeployment() {
+        return client
+                .apps()
+                .deployments()
+                .inNamespace(getNamespace())
+                .withName(getKeycloakName())
+                .get();
+    }
+
+    private Job buildJob(Container keycloakContainer, Volume secretVolume) {
+        return new JobBuilder()
+                .withNewMetadata()
+                .withName(getName())
+                .withNamespace(getNamespace())
+                .endMetadata()
+                .withNewSpec()
+                .withNewTemplate()
+                .withNewSpec()
+                .withContainers(keycloakContainer)
+                .addToVolumes(secretVolume)
+                .withRestartPolicy("Never")
+                .endSpec()
+                .endTemplate()
+                .endSpec()
+                .build();
+    }
+
+    private Volume buildSecretVolume() {
+        return new VolumeBuilder()
+                .withName(volumeName)
+                .withSecret(new SecretVolumeSourceBuilder()
+                        .withSecretName(secretName)
+                        .build())
+                .build();
+    }
+
+    private Job createImportJob() {
+        var keycloakContainer = buildKeycloakJobContainer();
+        var secretVolume = buildSecretVolume();
+        var importJob = buildJob(keycloakContainer, secretVolume);
+
+        return importJob;
+    }
+
+    private Container buildKeycloakJobContainer() {
+        var keycloakContainer =
+            this
+                .existingDeployment
+                .getSpec()
+                .getTemplate()
+                .getSpec()
+                .getContainers()
+                .get(0);
+
+        var importMntPath = "/mnt/realm-import/";
+
+        var command = List.of("/bin/bash");
+
+        var commandArgs = List.of("-c",
+                "/opt/keycloak/bin/kc.sh build && " +
+                    "/opt/keycloak/bin/kc.sh import --file='" + importMntPath + getRealmName() + "-realm.json' --override=true");
```

Comment:
I have been going forth and back in this decision, let's discuss what's the desired behaviour.
Maybe I should add this as a parameter in the CR?

Reply:
added to the CR and defaulted to `false`

Reply:
I believe this should not be configurable. What is the use case you have in mind when to use `override=true`?

Reply:
Updates, e.g.:

- import a default realm
- add a user
- re-import to add the user

I think that, since this is already a Keycloak feature, we can simply expose it to the users.

Reply:
The import CRD is not designed for this. In your example, how would you re-trigger the import? If it's marked as done, reconciliation loop won't re-run the job. Please note that users should not update the status field. Re-creating the CR is not IMHO very cloud-native.

It's not meant to fully suplement the CRDs from the old operator, rather to offers a reasonable alternative.

Reply:
If we don't expose this field people would need to perform manual operations on Keycloak to e.g. remove an old realm before importing.
I do believe that this is even worst than having the option.

But, this is a prototype, and you are arguing toward the safest path ... I'm ok with it.

Reply:
👍  Please mark this as resolved.